olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.cpp

```diff
@@ -288,6 +326,91 @@ model::Partition PartitionsRepository::PartitionFromSubQuad(
   return ret;
 }
 
+PartitionsResponse PartitionsRepository::GetAggregatedTile(
+    const client::HRN& catalog, const std::string& layer,
+    client::CancellationContext context, const TileRequest& request,
+    boost::optional<int64_t> version, const client::OlpClientSettings& settings,
+    bool aggregated) {
+  const auto fetch_option = request.GetFetchOption();
+  const auto& tile_key = request.GetTileKey();
+
+  // Download quad tree index depth 4 or load from cache
+  if (fetch_option != OnlineOnly && fetch_option != CacheWithUpdate) {
+    repository::PartitionsCacheRepository repository(
+        catalog, settings.cache, settings.default_cache_expiration);
+
+    // handle tiles from level < kAggregateQuadTreeDepth
+    auto max_depth =
+        std::min<int>(int(tile_key.Level()), kAggregateQuadTreeDepth);
+    QuadTreeIndex cached_tree;
+    // search in the cache for any quad tree covering the tile
+    for (auto i = max_depth; i >= 0; --i) {
+      const auto& root_tile_key = tile_key.ChangedLevelBy(-i);
+      cached_tree = repository.Get(layer, root_tile_key,
+                                   kAggregateQuadTreeDepth, version);
+      if (!cached_tree.IsNull()) {
+        OLP_SDK_LOG_DEBUG_F(kLogTag, "cache QuadTree (%s, %" PRId32 ") found!",
```

Comment:
I would prefer you follow same approach as below methods:

`GetPartitionForVersionedTile found in cache, hrn='%s', key='%s'`